src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -551,8 +550,8 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         LOG_TRACE(log, "Running 'analyze' second time");
 
         /// Reuse already built sets for multiple passes of analysis
-        subquery_for_sets = std::move(query_analyzer->getSubqueriesForSets());
-        prepared_sets = query_info.sets.empty() ? query_analyzer->getPreparedSets() : query_info.sets;
+        subquery_for_sets = std::make_shared<SubqueriesForSets>(std::move(query_analyzer->getSubqueriesForSets()));
```

Comment:
`subquery_for_sets` is always initialized, so I'd prefer to use here move-assignment without new memory allocation.